pyhmy/validator.py

```diff
@@ -0,0 +1,651 @@
+        except (RPCError, RequestsError, RequestsTimeoutError) as e:
+            raise InvalidValidatorError(5, f'Error requesting validator information') from e
+        try:
+            validator_info = get_validator_information(self._address, endpoint, timeout)
+        except (RPCError, RequestsError, RequestsTimeoutError) as e:
+            raise InvalidValidatorError(5, f'Error requesting validator information') from e
+
+        # Skip additional sanity checks when importing from chain
+        try:
+            info = validator_info['validator']
+            self._name = info['name']
+            self._identity = info['identity']
+            self._website = info['website']
+            self._details = info['details']
+            self._security_contact = info['security-contact']
+
+            self._min_self_delegation = convert_atto_to_one(info['min-self-delegation']).normalize()
+            self._max_total_delegation = convert_atto_to_one(info['max-total-delegation']).normalize()
+            self._amount = Decimal(0) # Since validator exists, set initial delegation to 0
+
+            self._max_rate = Decimal(info['max-rate']).normalize()
+            self._max_change_rate = Decimal(info['max-change-rate']).normalize()
+            self._rate = Decimal(info['rate']).normalize()
+        except KeyError as e:
+            raise InvalidValidatorError(5, f'Error importing validator information from RPC result') from e
+
+
+    def export(self) -> dict:
+        """
+        Export validator information as dict
+
+        Returns
+        -------
+        dict
+            Dictionary representation of validator
+        """
+        info = {
+            "validator-addr": self._address,
+            "name": self._name,
+            "website": self._website,
+            "security-contact": self._security_contact,
+            "identity": self._identity,
+            "details": self._details,
+            "amount": self._inital_delegation,
+            "min-self-delegation": self._min_self_delegation,
+            "max-total-delegation": self._max_total_delegation,
+            "rate": self._rate,
+            "max-rate": self._max_rate,
+            "max-change-rate": self._max_change_rate
+        }
+        return info
+
+    def export_json(self, pretty=False) -> str:
+        """
+        Export validator information as JSON string
+
+        Parameters
+        ----------
+        pretty: :obj:`bool`, optional
+            True to return pretty print json string
+
+        Returns
+        -------
+        str
+            JSON representation of validator
+
+        See also
+        --------
+        export
+        """
+        info = self.export()
+        for key, value in info.items():
+            if isinstance(value, Decimal):
+                info[key] = str(value)
+        if pretty:
+            return json.dumps(info, indent=4)
+        return json.dumps(info)
```

Comment:
Consider making this part of the `__str__` method, so if someone wants to make it pretty they could do something like:
```python
import json
val = Validator(...)
print(json.dumps(str(val), indent=2))
```

Having it as str would also help with debugging in Interpreter.